### PR TITLE
[WFLY-7137] KeyStore attribute "required" fix

### PR DESCRIPTION
**Pull request by honza889**
https://issues.jboss.org/browse/WFLY-7137

### CODE REVIEW

File: src/main/java/org/wildfly/extension/elytron/KeyStoreService.java [hunk 1]
```diff
@@ -114,6 +114,13 @@ public void start(StartContext startContext) throws StartException {
             synched = System.currentTimeMillis();
             try (InputStream is = resolvedPath != null ? new FileInputStream(resolvedPath) : null) {
```
Comment (darranl) [inline src/main/java/org/wildfly/extension/elytron/KeyStoreService.java]: Am not sure this is really the correct fix here - as resolvedPath is a File object I think we should perform required validation in advance and adjust the InputStream accordingly.

i.e. set the InputStream to null if the resolvedPath does not exist and required==false.  If required==true and it does not exist we can report an early error.

The IOException caught below could be caused by a number of issues so really we don't want to mix in the existence checking there.

**Comment by honza889** [comment]
Updated by notes.

**Comment by darranl** [comment]
Looks good.

**Comment by darranl** [comment]
Looks like this one needs a rebase.

**Comment by honza889** [comment]
Rebased.

**Comment by darranl** [comment]
Sorry @honza889 looks like we now have a test failure: -

```
testKeystoreService(org.wildfly.extension.elytron.KeyStoresTestCase)  Time elapsed: 0.018 sec  <<< ERROR!
java.lang.UnsupportedOperationException: null
	at org.jboss.as.controller.ObjectTypeAttributeDefinition.parse(ObjectTypeAttributeDefinition.java:96)
	at org.jboss.as.controller.SimpleAttributeDefinition.parseAndSetParameter(SimpleAttributeDefinition.java:186)
	at org.wildfly.extension.elytron.TlsParser.readKeyStore(TlsParser.java:475)
	at org.wildfly.extension.elytron.TlsParser.readKeyStores(TlsParser.java:436)
	at org.wildfly.extension.elytron.TlsParser.readTls(TlsParser.java:123)
	at org.wildfly.extension.elytron.ElytronSubsystemParser.readElement(ElytronSubsystemParser.java:131)
	at org.wildfly.extension.elytron.ElytronSubsystemParser.readElement(ElytronSubsystemParser.java:73)
	at org.jboss.staxmapper.XMLMapperImpl.processNested(XMLMapperImpl.java:110)
	at org.jboss.staxmapper.XMLExtendedStreamReaderImpl.handleAny(XMLExtendedStreamReaderImpl.java:69)
	at org.jboss.as.subsystem.test.TestParser.readElement(TestParser.java:103)

```

**Comment by honza889** [comment]
Fixed.

**Pull request merged**